src/ts/utils/roam.ts

```diff
@@ -58,7 +63,80 @@ export const Roam = {
         await Keyboard.pressEnter(15);
         await Keyboard.pressEnter();
         document.execCommand('paste')
+    },
+
+    moveCursorToStart() {
+        this.applyToCurrent(
+            node =>
+                new RoamNode(
+                    node.text,
+                    // new Selection(node.text.length * 2, node.text.length * 2)
+                    new Selection(0,0)
+                ))
+    },
+
+    moveCursorToEnd() {
+        this.applyToCurrent(
+            node =>
+                new RoamNode(
+                    node.text,
+                    new Selection(node.text.length * 2, node.text.length * 2)
+                ))
+    },
+
+    writeText(text: string) {
+        this.applyToCurrent(node => 
+            new RoamNode(text, node.selection));
+    },
+
+    async createSiblingAbove() {
+        this.moveCursorToStart();
+        const isEmpty = !this.getActiveRoamNode()?.text;
+        await Keyboard.pressEnter(20);
+        if (isEmpty) {
+            await Keyboard.simulateKey(38,20); //Up Arrow
```

Comment:
create functions for arrow keys, or at least constants with meaningful names instead of  comments